lib/factories/default-eventgate.js

```diff
@@ -168,34 +183,135 @@ function makeValidate(options, logger) {
 }
 
 /**
- * Creates a function that returns a connected Kafka Producer's produce function,
- * suitable for passing to EventGate as the produce function argument.
+ * Creates a function that writes events to output_path.
+ * @param {Object} options
+ * @param {Object} options.output_path
+ * @param {Object} logger
+ * @param {Object} metrics
+ * @return {EventGate~produce} (event, context) => Promise<event>
+ */
+function makeFileProduce(options, logger, metrics) {
+    const fs = require('fs');
+    const {
+        once
+    } = require('events');
+
+    const writeOptions = { flags: 'as' };
+    if (options.output_path === 'stdout') {
+        // If fd is set, createWriteStream will ignore output_path.
+        writeOptions.fd = process.stdout.fd;
+    }
+    const outputStream = fs.createWriteStream(options.output_path || './output.json', writeOptions);
+
+    return async (event, context = {}) => {
+        const serializedEvent = Buffer.from(JSON.stringify(event) + '\n');
+        if (!outputStream.write(serializedEvent)) {
+            await once(outputStream, 'drain');
+        }
+    };
+}
+
+/**
+ * Creates a function that returns function that produces events to Kafka.
  * NOTE: This function uses undefined Kafka key and partition when producing.
  * If you need to set key and/or partition, you should make your own produce function.
  * @param {Object} options
  * @param {string} options.stream_field
  *      Used to construct the kafka topic
 * @param {string} options.schema_uri_field
  *      Used to construct the kafka topic if stream_field is not set.
- * @param {string} options.key_field
- *      If given, the event's 'key' will be extracted from the event at this dotted object path.
- * @param {string} options.partition_field
- *      If given, the event's 'partition' will be extracted from the event at this
- *      dotted object path.
- * @param {KafkaProducer} producer
- *      Connected node-rdkafka Producer.
- * @return {EventGate~produce} (event, context) => Promise<event>
+ * @param {Object} options.kafka.conf
+ *      node-rdkafka KafkaProducer configuration
+ * @param {Object} options.kafka.topic_conf
+ *      node-rdkafka KafkaProducer topic configuration
+ * @param {Object} logger
+ * @param {Object} metrics
+ *      service-runner metrics object, used if statistics.interval.ms is set
+ *      to collect rdkafka metrics.
+ * @return {Promise<{EventGate~produce}>} (event, context) => Promise<event>>
  */
-function makeProduce(options, producer) {
+async function makeKafkaProduce(options, logger, metrics) {
+    const kafka = require('node-rdkafka-factory');
```

Comment:
maybe wrap in try-catch and re-throw with a nice error message and a hint to install the dependency?

Reply:
I think in this case the Error that is thrown is decent (and standard) when a dependency is not installed.